geode-core/src/test/java/org/apache/geode/internal/cache/ClientServerTransactionDUnitTest.java

```diff
@@ -4065,4 +4065,157 @@ private void doJTATx2(String regionName, CountDownLatch latch1, CountDownLatch l
       Assert.fail("Unexpected exception while doing JTA Transaction2 ", e);
     }
   }
+
+  @Test
+  public void testPartitionMessageSetsClientMemberIdAsTxMemberId() {
```

Comment:
This test class is now over 4000 lines long. Should we start considering breaking this apart? It seems like too much state needs to be passed around because these tests cannot have any state at the test instance level, due to the varied nature of the numerous tests in the file.

Reply:
Yes, I agree. We may need to add a new JIRA to break down the test.